src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/DeliverySessionService.kt

```diff
@@ -145,13 +150,18 @@ class DeliverySessionService(
       SERVICE_DELIVERY,
       npsOfficeCode
     )
+    if (appointmentTime.isBefore(OffsetDateTime.now()) && attended == null) {
+      throw IllegalArgumentException("Appointment feedback must be provided for appointments in the past.")
+    }
     appointmentToSchedule.appointmentTime = appointmentTime
     appointmentToSchedule.durationInMinutes = durationInMinutes
     appointmentToSchedule.deliusAppointmentId = deliusAppointmentId
     appointmentRepository.saveAndFlush(appointmentToSchedule)
     appointmentService.createOrUpdateAppointmentDeliveryDetails(appointmentToSchedule, appointmentDeliveryType, appointmentSessionType, appointmentDeliveryAddress, npsOfficeCode)
     deliverySession.appointments.add(appointmentToSchedule)
-    return deliverySessionRepository.saveAndFlush(deliverySession)
+    return deliverySessionRepository.saveAndFlush(deliverySession).also {
+      setAttendanceAndBehaviourIfHistoricAppointment(deliverySession, appointmentToSchedule, appointmentTime, attended, additionalAttendanceInformation, behaviourDescription, notifyProbationPractitioner, scheduledBy)
```

Comment:
this is worth a comment as to why it appears here. 

also, does the appointment need saving again a the end here?